server/aws-lsp-codewhisperer/src/language-server/securityScan/codeWhispererSecurityScanServer.ts

```diff
@@ -192,6 +195,12 @@ export const SecurityScanServerToken =
                 logging.log(`Security scan failed. ${error}`)
                 securityScanTelemetryEntry.result = 'Failed'
                 const err = getErrorMessage(error)
+                const authFollowType = getAuthFollowUpType(error)
+                if (authFollowType == 're-auth') {
+                    throw new ResponseError(LSPErrorCodes.RequestFailed, err, {
```

Comment:
this would throw an error back to the toolkit and not hit L205-207 which we want to send back incase of failures.
I would recommend changing the return payload to include the responseError object you created.
```
var exception = getException(error) // this sends back the response error in case or re-auth or the base `error` as is
  return {
                    status: 'Failed',
                    error: errorrMessage,
                    exception: exception
                } as SecurityScanResponse
```

Reply:
Also any reason we cannot create the AmazonQServiceAuthenticationExpired exception here instead of a new Response Error? This will allow us to possibly create a reusable utility call that creates this exception and use it across various features i.e in this class and security scan server